core/scripts/local/CalculateContractBytecode.js

```diff
@@ -0,0 +1,24 @@
+// This simple script tells you how big your contract byte code is and how much you have until you exceed
+// the current block limit as defined by EIP170.
+
+const argv = require("minimist")(process.argv.slice(), { string: ["contract"] });
+const contractName = argv.contract;
+
+if (!contractName) {
+  console.log("Please enter the contract name as a parameter as `--contract <name>`.");
+  return;
+}
+var child = require("child_process").exec("truffle compile");
+child.stdout.pipe(process.stdout);
+child.on("exit", function() {
```

Comment:
Not a strong opinion, just want to hear your thoughts:

WDYT about wrapping this in a bash script that does the compilation instead of doing all this child process stuff? Something like:
```
# Usage example:
# ./scripts/CalculateContractBytecode.sh Voting

$(npm bin)/truffle compile
node core/scripts/local/CalculateContractBytecode.js --contract $1
```

Another option might be doing *everything* in bash, but that may be a pretty annoying. A one-liner to grab the length of the `bytecode` field in bash using the `jq` package if you were to choose to go in that direction:
```
cat build/contracts/Voting.json | jq -r '.bytecode' | wc -c
```

My only slight bias against messing with child processes is that I think they can be a bit error prone since the language is attempting to directly interface with the user's shell environment.

Reply:
I originally did not use the child process method and tried a few other ways of doing it but ultimately this was the least clunky way of executing it _from_ node. A bash script is probably more reliable but would result in needing two scripts. Doing it in all bash is a pain. For example the code example you gave with `jq` assumes that that has been installed by on the users machine; I dont have that installed and so it did not work. 

However, if you have have installed the node modules then you will have truffle installed. I changed it to rather than using the system truffle version to rather use this:
```
$(npm bin)/truffle compile

```
which I think will make it more resilient to these cases.